apis/itemsetapi/itemsetapi.yaml

```diff
@@ -0,0 +1,840 @@
+      responses:
+        '200':
+          description: >-
+            SUCCESS. The **Publish ItemSet** operation was successful!
+          schema:
+            $ref: '#/definitions/ItemSetCreateResponse'
+        '400':
+          description: >-
+            'BAD REQUEST. The **Publish ItemSet** operation failed. You may have missed input for a mandatory parameter.'
+          schema:
+            $ref: '#/definitions/ItemSetCreateErrorResponse'
+        '500':
+          description: >-
+            'INTERNAL SERVER ERROR. We track these errors automatically and try to set it right at the earliest. Try refreshing the page. If the problem persists contact us at info@sunbird.org.'
+          schema:
+            $ref: '#/definitions/ItemSetCreateErrorResponse'
+      security:
+        - bearer: []
+      x-unitTests:
+        - request:
+            method: POST
+            uri: /itemset/v1/publish/{ItemSet_Id}
+            headers:
+              Content-Type: application/json
+              X-Authenticated-Userid: '{{user_id}}'
+              Authorization: 'Bearer {{api_key}}'
+            body: |-
+              {
+                "request": {
+                  "itemset": {
+                    "lastPublishedBy": "89cf1a7e-dfd3-46c9-a428-d37e9a2bc001"
+                  }
+                }
+              }
+          expectedResponse:
+            x-allowExtraHeaders: true
+            x-bodyMatchMode: NONE
+            x-arrayOrderedMatching: false
+            x-arrayCheckCount: false
+            x-matchResponseSchema: true
+            headers: {}
+          x-testShouldPass: true
+          x-testEnabled: true
+          x-testName: Publish ItemSet
+          x-testDescription: ''
+      x-operation-settings:
+        CollectParameters: false
+        AllowDynamicQueryParameters: false
+        AllowDynamicFormParameters: false
+        IsMultiItemSetStreaming: false
+  /retire/{ItemSet_ID}:
+    delete:
+      description: >-
+        This API is associated with retiring the itemset on the Sunbird Platform.
+          - The endpoint for **Retire ItemSet** is `/retire/{ItemSet_ID}`
+          - The fields marked with an asterisk (*) are mandatory. They cannot be null or empty.
+
+      summary: Retire ItemSet
+      tags:
+        - ItemSet APIs
+      operationId: Retire ItemSet
+      produces:
+        - application/json
+      parameters:
+        - in: path
+          name: ItemSet_ID
+          type: string
+          required: true
+          description: 'Please append a valid ItemSet Id To the Request URL'
+        - name: Body
+          in: body
+          required: true
+          description: >-
+           The body parameter refers to the format of the request, the request body (payload) takes the following media types to make the payload:'
+            
+            - JSON 
+            - XML data
+          
+            The body is the representation of the resource object for deleting item set
```

Comment:
does retiring the ItemSet = deleting the ItemSet? this name for this endpoint is retire , not delete. Hence please make it consistent.